### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by mverzilli**
Basic documentation for forwarding logs feature.

localhost_8080_defender_logging html (1)

### CODE REVIEW

**Comment by zeljkoX** [comment]
@mverzilli This is ready for review. I cannot add you as a reviewer for some reason as you are creator of this PR.

**Comment by mverzilli** [comment]
On it!

File: docs/modules/ROOT/pages/logging.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
+= Logging
+
+Defender generates log trail of every potentially relevant event in the system.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:3]:
```suggestion
Defender generates log trails of every potentially relevant event in the system.
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+== Log Forwarding
+
+Generated logs can be forwarded to Datadog and Splunk or any other service that supports API Key authorization.
+
+This is a premium feature, if you are interested in it let us know by filling in this https://zpl.in/defender-inquiry[form, window=_blank].
+
+== Setup Log Forwarding Destination
+
+Log Forwarding configuration page can be accessed from the User Menu or by following this https://defender.openzeppelin.com/#/log-forwarding[link,window=_blank].
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:13]:
```suggestion
To set up a log forwarding destination, open the User Menu at the top right corner of Defender or following this https://defender.openzeppelin.com/#/log-forwarding[click here,window=_blank].
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+image::log-forwarding-add.png[Add Forwarding Destination, 600]
+
+Form for creating a new Destination can be accessed by following `Add Forwarding Destination` button.
+
+image::log-forwarding-form.png[Add Log Destination, 600]
+
+Form fields:
+
+* *URL* field is a required field. All logs are forwarded to this URL address by using HTTP POST method.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:23]:
```suggestion
* *URL* field is a required field. All logs are forwarded to this URL address using HTTP POSTs.
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+* *API Header Name* is optional. This is the name of the request header that contains the API Key value. For most log management services API Header Name needs to be set.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:25]:
```suggestion
* *API Header Name* is optional. This is the name of the request header that contains the API Key value. Most log management services require it. Please refer to your log management service documentation to determine if you need it. 
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+* *API Key* is an optional field. API Key is sent with every request in order to authenticate it. For most log management services API Key needs to be set.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:27]:
```suggestion
* *API Key* is an optional field. API Key is sent with every request for authentication purposes. Most log management services require it. Please refer to your log management service documentation to determine if you need it. 
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+* *Log Types* field enables filtering only certain types of logs for forwarding.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:29]:
```suggestion
* *Log Types* lets you specify which subset of Defender generated logs you want to have forwarded.
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+NOTE: In the next section we will cover how to setup Log Forwarding with Splunk and Datadog but it is worth noting that Log Forwarding works with any other service that supports API Key authorization.
+
+=== Splunk
+
+Forwarding logs to Splunk is done by using Splunk HEC(HTTP Event Collector).
+Documentation for setting up logging with Splunk HEC can be found on this https://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector[link,window=_blank].
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:36]:
```suggestion
Documentation for setting up logging with Splunk HEC can be found https://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector[here,window=_blank].
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+* *API Key*: `Splunk xxxxxxxxxxxxxxxxxxxxxxxxxxx`
+
+NOTE: `URL` value is dynamic as URL includes account username.
+
+NOTE: `API Key` should contain `Splunk` prefix.
+
+
+=== Datadog
+
+Documentation for setting logging on Datadog can be found on this https://docs.datadoghq.com/logs/[link,window=_blank].
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:55]:
```suggestion
Documentation for setting up logging on Datadog can be found https://docs.datadoghq.com/logs/[here,window=_blank].
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+Example:
+
+* *URL*: `\https://http-intake.logs.datadoghq.com/api/v2/logs`
+
+* *API Header Name*: `DD-API-KEY`
+
+* *API Key*: `xxxxxxxxxxxxxxxxxxxxxxxxxxx`
+
+NOTE: Datadog uses different sites around the world. For EU server `URL` field value should be https://http-intake.logs.datadoghq.eu/api/v2/logs
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:66]:
```suggestion
NOTE: Datadog uses different sites around the world. For example, if you are relying on an EU server the `URL` field value should be https://http-intake.logs.datadoghq.eu/api/v2/logs
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+NOTE: `API Key` value can be obtained from Datadog site by opening `Logs` section from the left menu.
+`Cloud` section and `AWS` provider needs to be selected.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:69]:
```suggestion
Go to `Cloud` section and select `AWS` provider.
```

File: docs/modules/ROOT/pages/logging.adoc [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+After following steps, the 'API Key` value is displayed in the bottom section of the page.
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:70]:
```suggestion
After following those steps, the 'API Key` value is displayed in the bottom section of the page.
```

**Comment by zeljkoX** [comment]
@mverzilli thanks for your great feedback, as always.

Just to verify again does "this Click here" looks ok in combination together?
Screenshot 2022-03-24 at 11 57 14

**Comment by mverzilli** [comment]
> @mverzilli thanks for your great feedback, as always.
> 
> Just to verify again does "this click here" looks ok in combination together? Screenshot 2022-03-24 at 11 57 14

Good catch, yeah, should be "right corner of Defender or click here"

File: docs/modules/ROOT/pages/logging.adoc [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+== Log Forwarding
+
+Generated logs can be forwarded to Datadog and Splunk or any other service that supports API Key authentication.
+
+This is a premium feature, if you are interested in it let us know by filling in this https://zpl.in/defender-inquiry[form, window=_blank].
+
+== Setup Log Forwarding Destination
+
+To set up a log forwarding destination, open the User Menu at the top right corner of Defender or following this https://defender.openzeppelin.com/#/log-forwarding[click here,window=_blank].
```
Comment (mverzilli) [inline docs/modules/ROOT/pages/logging.adoc:13]:
```suggestion
To set up a log forwarding destination, open the User Menu at the top right corner of Defender or https://defender.openzeppelin.com/#/log-forwarding[click here,window=_blank].
```

**Pull request merged**